### PR TITLE
Set max age connection for built in plugins gRPC server

### PR DESCRIPTION
**Pull request by marcosy**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**
gRPC server configuration for builtin plugins

**Description of change**
In a large SPIRE deployment, I have noticed (after a couple of days) that some gRPC errors (`transport is closing`) start to appear in the SPIRE server log. Particularly when trying to get information from the datastore. This could lead the agent to fail the SVID rotation.

```
time="2019-09-15T10:50:05Z" level=error msg="failed to fetch bundle: rpc error: code = Unavailable desc = transport is closing" subsystem_name=node_api
```

I believe the problem is that someone (not sure if the platform or the OS) is force closing the gRPC connection used by the built-in plugins when the connection reaches a certain age. In my case, I have always seen this after ~2.5 days.

I propose to set values for `MaxConnectionAge` and `MaxConnectionIdle` instead of using the defaults (infinity). In this way, we can ensure the gRPC connection is closed/renewed properly at a given time interval.

After setting those values I did not see that error message any more.

### CODE REVIEW

**Comment by azdagron** [comment]
I'm really interested to know more details of what is happening to the connection under the covers. The builtin connections are done via `net.Pipe()` and are in-memory constructs only so it can't be the OS that is closing them. Maybe some interaction with the gRPC server/client setting deadlines? Did you happen to capture verbose GRPC logs to figure out what was happening?

I'm not opposed to taking the change if it fixes the problem, I'm just not personally clear on the details and would like to understand the situation better.

**Comment by marcosy** [comment]
> Did you happen to capture verbose GRPC logs to figure out what was happening?

I tried, but the deployment was very large and the average wait time to see the error was of a couple of days... that made the logs hard to follow. 
I'll try to reproduce this in a simpler environment and enable GRPC logs there to get more info.

**Comment by evan2645** [comment]
Hey @marcosy. I discussed this change with @azdagron last week. A few things we identified:
* We haven't been able to reproduce this bug with recent SPIRE builds
* It was encountered while using an older version of gRPC than we are using today
* This seems to be the right fix for problems related to TCP keep-alives, but we are not using TCP in this case
* We _think_ that we will re-dial if the connection is closed because this fix forces it to close every so often - and if that's the case, then the impact of this bug is limited to a log message and perhaps a failed RPC (and thus recoverable)
* This problem has only been reported once, and it's hard to know exactly how or why it was occurring

So... I think there's a little bit of hesitance to merge this without better understanding exactly what's going on, and being sure that it is a problem that still exists after the lib updates etc. Considering all of these things, and the age of this PR, how would you feel about closing it for now? We can always re-open it in the future if it pops up again.

**Comment by evan2645** [comment]
Gonna go ahead and close this - please let me know if you want to re-open it @marcosy

**Comment by marcosy** [comment]
Sorry @evan2645 ! I missed this comment. Closing this PR sounds totally reasonable to me.
